src/udev/udev-spawn.c

```diff
@@ -54,35 +56,54 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
         assert(fd == spawn->fd_stdout || fd == spawn->fd_stderr);
         assert(!spawn->result || spawn->result_len < spawn->result_size);
 
-        if (fd == spawn->fd_stdout && spawn->result) {
+        if (fd == spawn->fd_stdout && spawn->result && !spawn->truncated) {
+                /* When reading to the result buffer, use the maximum available size, to detect truncation. */
+                read_to_result = true;
```

Comment:
`read_to_result` is equivalent to `fd == spawn->fd_stdout && spawn->result && !spawn->truncated`, and it is used once, so looks needless.

Reply:
This is for readability. Writing the same condition after `read()` is not good.